plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineExecutionService.java

```diff
@@ -40,13 +43,25 @@ public PipelineExecutionService(PipelineStore store, ThreadPool threadPool) {
         this.threadPool = threadPool;
     }
 
-    public void execute(Data data, String pipelineId, Listener listener) {
+    public Pipeline getPipeline(String pipelineId) {
         Pipeline pipeline = store.get(pipelineId);
+
         if (pipeline == null) {
-            listener.failed(new IllegalArgumentException(LoggerMessageFormat.format("pipeline with id [{}] does not exist", pipelineId)));
-            return;
+            throw new IllegalArgumentException(LoggerMessageFormat.format("pipeline with id [{}] does not exist", pipelineId));
         }
 
+        return pipeline;
+    }
+
+    public void execute(Data data, String pipelineId, Listener listener) {
```

Comment:
yes, some of these changes were before a rapid refactor last night while speaking with Martijn. Not everything was cleaned up during that time. That may be why some things seem unused and comments may not make sense :)